src/parse.rs

```diff
@@ -474,6 +491,25 @@ impl Extensions {
     }
 }
 
+pub trait Num: Sized {
+    fn from_str(src: &str, radix: u32) -> ::std::result::Result<Self, ()>;
+}
+
+macro_rules! impl_num {
+    ($ty:ident) => {
+        impl Num for $ty {
+            fn from_str(src: &str, radix: u32) -> ::std::result::Result<Self, ()> {
```

Comment:
nit: maybe we can pass-through the error (`ParseIntError`) here?

Reply:
I cannot create the `Error` without line and column, and returning `ParseIntError` isn't better because I want to use `self.error()`.